command/init.go

```diff
@@ -121,13 +121,58 @@ func (c *InitCommand) RunContext(buildCtx context.Context, cla *InitArgs) int {
 			Getters:                   getters,
 		})
 		if err != nil {
-			c.Ui.Error(err.Error())
-			ret = 1
+			if pluginRequirement.Implicit {
+				msg := fmt.Sprintf(`
+Warning, at least one component used in your config file(s) has moved out of 
+Packer into the %q plugin.
+For that reason Packer init tried to install the latest version of the %s 
+plugin. Unfortunately, this failed :
+%s`,
+					pluginRequirement.Identifier,
+					pluginRequirement.Identifier.Type,
+					err)
+				c.Ui.Say(msg)
+			} else {
+				c.Ui.Error(err.Error())
+				ret = 1
+			}
 		}
 		if newInstall != nil {
+			if pluginRequirement.Implicit {
+				msg := fmt.Sprintf("Installed implicitly required plugin %s %s in %q", pluginRequirement.Identifier, newInstall.Version, newInstall.BinaryPath)
+				ui.Say(msg)
+
+				warn := fmt.Sprintf(`
+Warning, at least one component used in your config file(s) has moved out of 
+Packer into the %[2]q plugin.
+For that reason Packer implicitly required the installation of the latest
+version of the %[1]s plugin.
+Upon init, Packer will always fetch the latest possible version of implicitly
+required plugins and if a latest version is backward incompatible with your
+config file or your version of Packer: a build will fail. To avoid this, lock
+the plugin version by pasting the following to your config:
```

Comment:
I think this warning is too long and it will be longer if more the one plugin is implicitly required. I tried making it shorter without losing information. See if you like it. 
```suggestion
Warning! At least one component used in your config file(s) has moved out of 
Packer into the %[2]q plugin.
Upon init, Packer will always fetch the latest possible version of implicitly
required plugins. To avoid backward-incompatibilities, lock
the plugin version by pasting the following to your config:
```

Reply:
For this one I like to sort of explain the why because 'implicitly required plugin' is sort of our technical term and I think it is nice to explain it here 🤔 .

What do you think @nywilken & @SwampDragons ?

Reply:
I like the context on why here. An alternative could be to point to a document on the packer.io site. 

```
```suggestion
Warning, at least one component used in your config file(s) has moved out of 
Packer into the %[2]q plugin and is now being implicitly required. 
For more details on implicitly required plugins see https://packer.io/plugins/#required_plugins

To avoid any backward incompatible changes with your
config file you may want to lock the plugin version by pasting the following to your config:
```

If you would prefer to keep the text in the error message what about a slight reword on the last two sentences

```
Warning, at least one component used in your config file(s) has moved out of 
Packer into the %[2]q plugin.
For that reason Packer has implicitly required the installation of the latest
version of the %[1]s plugin.
Upon init, Packer will fetch the latest possible version of any implicitly
required plugin(s). To avoid any backward incompatible changes with your
config file you may want to lock the plugin version by pasting the following to your config:
```

Reply:
I think I like the idea to link to our docs, way better. ( way less spam in the terminal and once you've read that you know, so less text to ignore ).

Reply:
```suggestion
Warning, at least one component used in your config file(s) has moved out of 
Packer into the %[2]q plugin and is now being implicitly required. 
For more details on implicitly required plugins see https://packer.io/docs/commands/init#implicit-plugin-requirements

To avoid any backward incompatible changes with your
config file you may want to lock the plugin version by pasting the following to your config:
```